src/queries/role.js

```diff
@@ -40,7 +41,20 @@ function queryAllByRole(
       const isRoleSpecifiedExplicitly = node.hasAttribute('role')
 
       if (isRoleSpecifiedExplicitly) {
-        return matcher(node.getAttribute('role'), node, role, matchNormalizer)
+        const roleValue = node.getAttribute('role')
+        if (queryFallbacks) {
+          return roleValue
+            .split(' ')
+            .filter(Boolean)
+            .some(text => matcher(text, node, role, matchNormalizer))
+        }
+        // if a custom normalizer is passed then let normalizer handle the role value
+        if (normalizer) {
```

Comment:
Do we have a test specifically covering this case? We should.

Reply:
I thought it is covered in [text-matcher.js](https://github.com/testing-library/dom-testing-library/blob/a2c3472edad89c0701692fd516869b74db453014/src/__tests__/text-matchers.js#L220-L221).

If this is not the test case you're talking about then can you please guide me where to add the test case for it? That will be very helpful :)

Reply:
@kentcdodds ping you to remind again.

Reply:
Gotcha. That's fine. Thank you!